### PR TITLE
Reach 100% coverage

### LINKED ISSUE #40: 100% coverage
Reach and maintain 100% coverage. 

A single environment run (`tox -e coverage`) currently reports 96.04% coverage; coveralls.io combines to report 97% (but I don't know how it rounds).

There are a number of completely unused files that I suspect/hope can be removed.

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #40 

It's mostly pretty routine. A few commits might have some more implications

- https://github.com/zopefoundation/zope.security/commit/e2d292a3623dd989a0d004a9a2bacc0d24346ef8 removes unused files from tests/ (which shouldn't have been used by other projects, but 🤷‍♂️ )
- https://github.com/zopefoundation/zope.security/commit/b710673d9b05c9aab3100c2a2d8dc97c083c87f6 uses the correct super class in proxy.py everywhere. I couldn't figure out why it was using the "wrong" class unless it was leftover from previous refactoring. `super(PyProxyBase)` would be `AbstractPyProxyBase`, so it wasn't actually bypassing any behaviour.
- https://github.com/zopefoundation/zope.security/commit/b5f2d262639d8d1d24e09e4c275c7f84435f000d fixes the Python proxy implementation to not call the target's `__getattribute__`/`__getattr__` functions more than once when they raise an AttributeError, just like the C version. The previous code that would have called it twice wasn't covered by tests until I added one and discovered the difference.

### CODE REVIEW

File: src/zope/security/checker.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ class DuplicationError(Exception):
 if os.environ.get('ZOPE_WATCH_CHECKERS'):
     try:
         WATCH_CHECKERS = int(os.environ.get('ZOPE_WATCH_CHECKERS'))
-    except ValueError:
+    except ValueError: # pragma: no cover
```
Comment (mgedmin) [inline src/zope/security/checker.py]: FWIW flake8 (via pycodestyle) complains when you don't precede an inline comment with two or more spaces.

(Personally I hate that rule.)

File: src/zope/security/tests/test_checker.py [hunk 2]
```diff
@@ -1849,14 +1802,14 @@ def test_proxy(self):
...
-            #    self.assertTrue(removeSecurityProxy(proxy) is ob)
+            #    self.assertIs(removeSecurityProxy(proxy), ob)
             #
             #    checker = getChecker(proxy)
             #    self.assertTrue(checker is special,
             #                    checker.get_permissions)
             #
             #    proxy2 = checker.proxy(proxy)
-            #    self.assertTrue(proxy2 is proxy, [proxy, proxy2])
+            #    self.assertIs(proxy2, proxy, [proxy, proxy2])
```
Comment (mgedmin) [inline src/zope/security/tests/test_checker.py]: Just delete commented code?

File: src/zope/security/tests/test_proxy.py [hunk 3]
```diff
@@ -1300,7 +1307,46 @@ def test___getslice___w_checker_allows(self):
         checker = DummyChecker()
         proxy = self._makeOne(target, checker)
         self.assertEqual(proxy[1:3], [1, 2])
-        self.assertEqual(checker._checked, '__getslice__')
+        self.assertEqual(checker._checked, self.getslice)
+
+    @_skip_if_not_Py2
+    def test___getslice___error_propagates(self):
+        # This is currently broken on Python 3 because
+        # https://github.com/zopefoundation/zope.proxy/issues/21
```
Comment (mgedmin) [inline src/zope/security/tests/test_proxy.py]: The fix for that one is merged already.

File: src/zope/security/tests/test_proxy.py [hunk 4]
```diff
@@ -1300,7 +1307,46 @@ def test___getslice___w_checker_allows(self):
...
+        with self.assertRaisesRegexp(Missing,
+                                     self.getslice):
+            proxy[1:2]
+
+        self.assertEqual(checker._checked, self.getslice)
+
+    @_skip_if_not_Py2
+    def test___getslice___dne_uses_getitem(self):
+        # This is currently broken on Python 3 because
+        # https://github.com/zopefoundation/zope.proxy/issues/21
```
Comment (mgedmin) [inline src/zope/security/tests/test_proxy.py]: Same.

File: src/zope/security/tests/test_proxy.py [hunk 5]
```diff
@@ -1331,6 +1377,41 @@ def _try():
...
+        proxy = self._makeOne(target, checker)
+        with self.assertRaisesRegexp(Missing,
+                                     self.setslice):
+            proxy[1:2] = 1
+
+        self.assertEqual(checker._checked, self.setslice)
+
+    def test___setslice___dne_uses_getitem(self):
+        # This is currently broken on Python 3 because
+        # https://github.com/zopefoundation/zope.proxy/issues/21
```
Comment (mgedmin) [inline src/zope/security/tests/test_proxy.py]: Same.

File: src/zope/security/checker.py (hunk 1 above)
Comment (jamadden) [reply src/zope/security/checker.py]: Yeah, I hate that rule too, and I have it turned off. It's [not](https://github.com/zopefoundation/zope.security/blame/master/src/zope/security/metaconfigure.py#L50) [like](https://github.com/zopefoundation/zope.security/blame/master/src/zope/security/metaconfigure.py#L52) [anyone](https://github.com/zopefoundation/zope.security/blame/master/src/zope/security/checker.py#L231) [actually](https://github.com/zopefoundation/zope.security/blame/master/src/zope/security/checker.py#L244) [does that](https://github.com/zopefoundation/ZODB/blame/master/src/ZODB/BaseStorage.py#L77) [consistently](https://github.com/zopefoundation/ZODB/blame/master/src/ZODB/BaseStorage.py#L78). 😉

File: src/zope/security/tests/test_checker.py (hunk 2 above)
Comment (jamadden) [reply src/zope/security/tests/test_checker.py]: Good idea. Will do.

**Comment by jamadden** [comment]
Thank you!

**Pull request merged**